build/lib/src/state/filesystem_cache.dart

```diff
@@ -129,4 +236,40 @@ class InMemoryFilesystemCache implements FilesystemCache {
     _stringContentCache[id] = result;
     return result;
   }
+
+  @override
+  void writeAsString(
+    AssetId id,
+    String contents, {
+    Encoding encoding = utf8,
+    required void Function() writer,
+  }) {
+    _stringContentCache[id] = contents;
```

Comment:
should there be some handling here of non-utf8 encoding? (I seem to recall a comment from some point about only saving utf8-encoded strings, also above we only read from _stringContentCache if using utf8)

Reply:
Yes, this was incorrect; fixed and added test coverage.

Could you take another look, please?